### PR TITLE
try qt6-main

### PR DESCRIPTION
**Pull request by looooo**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by looooo** [comment]
for osx-arm64 add this to the build-script:
```
if [[ "${target_platform}" =~ osx-arm64 ]]; then
    rm -f "${PREFIX}/lib/qt6/moc"
    ln -s "${BUILD_PREFIX}/lib/qt6/moc" "${PREFIX}/lib/qt6/moc"
    
    # Additional debugging information
    echo "Adjusted Qt tools for osx-arm64 with build variant qt6"
    echo "Removed: ${PREFIX}/lib/qt6/moc"
    echo "Linked to: ${BUILD_PREFIX}/lib/qt6/moc"
else
    echo "Skipping Qt tools adjustment. Target platform: ${target_platform}, Build variant: $build_variant"
fi
```

**Comment by looooo** [comment]
@conda-forge/core `build: -skip`is here for me not working. any ideas? I would like to keep the build matrix small and only build for py>311
additional the osmesa and egg variants are not needed as these builds are not qt dependent (TODO).

also I would like to know if we can upload with the label qt6 (currently it is set to vtk_qt6) but I guess having a qt6 label is more convenient as this can be used for more packages...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # otherwise it won't be correct in the final build.
 
 package:
-  name: vtk
+  name: vtk_qt6
```
Comment (hmaarrfk) [inline recipe/meta.yaml:19]: Is it safe to install both vtk and vtk_qt6 at the same time? If not you will need to enforce that with a mutually exclusive package. Is uggest you leave the package name the samr

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml]: sorry for the noise on this one, this was the meta package name.

**Comment by hmaarrfk** [comment]
can you rebase and remove the rerendering comits

**Comment by looooo** [comment]
@hmaarrfk what do you think about adding a qt6 label to the packages and use a new branch?

**Comment by hmaarrfk** [comment]
why? just rip off the bandaid tbh, it is very confusing to constantly tell people that they need to use a specific label or branch.

If you want to tell people to use a specific label, then you might as well just build our own channel (essentially equivalent) and keep it as updated as you want. It works on a small scale.


I think a better solution would be to update to 9.3 and then users can pin to to 9.2 for "qt5" and "9.3" for qt6.

Qt6, with the exception of qt-webengine, has been doing quite well.

**Comment by looooo** [comment]
> why? just rip off the bandaid tbh, it is very confusing to constantly tell people that they need to use a specific label or branch.

I just wanted to not introduce troubles for packages depending on qt5 and vtk. But I am happy to have it directly in conda-forge. 

> I think a better solution would be to update to 9.3 and then users can pin to to 9.2 for "qt5" and "9.3" for qt6.

sounds good to me.

**Comment by hmaarrfk** [comment]
I don't really know a way to avoid the troubles.

My attempt is to make an easy version cutoff that people that need old can still use. 

It is really hard to use build strings for this since build strings are somewhat adhoc

**Comment by looooo** [comment]
```
The reported errors are:
- Encountered problems while solving:
-   - package vtk-base-9.3.0-osmesa_py312h1234567_100 requires libxml2 >=2.11.6,<2.12.0a0, but none of the providers can be installed
```

**Comment by hmaarrfk** [comment]
I've had alot of trouble with the libxml2122 migration
https://github.com/conda-forge/qt-main-feedstock/pull/216/files

I've been going through the graph and doing it myself
https://conda-forge.org/status/#libxml2212

it seems that it is now vtk's turn.

**Comment by looooo** [comment]
@conda-forge-admin, please rerend

**Comment by looooo** [comment]
> I've had alot of trouble with the libxml2122 migration https://github.com/conda-forge/qt-main-feedstock/pull/216/files
> 
> I've been going through the graph and doing it myself https://conda-forge.org/status/#libxml2212
> 
> it seems that it is now vtk's turn.

seems like this pr is using libxml2 2.12.4, the pinning is:
```
libxml2:
- '2'
```


the failing build has error messages like this:
```
/home/conda/feedstock_root/build_artifacts/vtk_1706771685165/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include/GL/osmesa.h:323:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'OSMesaPostprocess'
```

**Comment by hmaarrfk** [comment]
we loosed libxml2 due to the fact that we believe it now better maintains abi compatibility accross v2.

It is unclear from the short error message that you have that the error is related to the libxml2.

**Comment by looooo** [comment]
> we loosed libxml2 due to the fact that we believe it now better maintains abi compatibility accross v2.

thanks, good to know.

Any ideas what to do with the failing build? If not, is it possible to merge also with the failing build?

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
it seems that the failing build is a mesalib build. 

was it working before? mesalib 24 just got released, so maybe there is some incompatibility.

**Comment by adrianinsaval** [comment]
@looooo could you check your personal repo's setting? it's uploading stuff to the freecad repo in anaconda which then breaks the freecad weekly builds https://cirrus-ci.com/task/4937101846773760?logs=osx_build#L1012

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by looooo** [comment]
@conda-forge/core is this rerendering issue knowen?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -32,13 +32,15 @@ source:
 build:
   number: {{ build }}
   string: {{ build_string }}
+  skip: true  # [py<312]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:35]: You really want to skip all but python 3.12???

Comment (looooo) [reply recipe/meta.yaml:35]: This line should skip all python versions smaller than 3.12. Just to prevent building the whole build matrix.

Comment (looooo) [reply recipe/meta.yaml:35]: this pre-rendering issue also occurs if I use the conga-forge/main branch for the vtk-feedstock.

Comment (hmaarrfk) [reply recipe/meta.yaml:35]: yeah i saw some other strange rendering things yesterday... i had to use the bot and not local.

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (looooo) [reply recipe/meta.yaml:35]: it happens to me on osx-arm, linux-aarch64 and with the bot.

Comment (hmaarrfk) [reply recipe/meta.yaml:35]: Yeah. I noticed. I don't have a clear answer

Comment (looooo) [reply recipe/meta.yaml:35]: conda-smithy 3.32.0 worked for me by not doing the version check. Version 3.33 and 3.34 have this issue.

**Comment by looooo** [comment]
ppc64le fails with:
```
Encountered problems while solving:
  - nothing provides requested proj 9.4.0.*
```

**Comment by hmaarrfk** [comment]
maybe you have to skip it:
https://github.com/conda-forge/proj.4-feedstock/pull/143

**Comment by hmaarrfk** [comment]
out of curiosity, why do you not want to build ffmpeg 6?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -32,13 +32,16 @@ source:
...
```
Comment (hmaarrfk) [inline recipe/meta.yaml:35]:
```suggestion
  skip: true  # [ffmpeg == 6]
  skip: true  # [py<312]
```

I think this should skip ffmpeg6 for you, but you should be able to build with both.

**Comment by looooo** [comment]
> out of curiosity, why do you not want to build ffmpeg 6?

keep the build-matrix smaller. My goal is to provide freecad builds for qt6. I think it's best to do this first via the anaconda.org freecad organization. I have the fear that qt6 is making things a little bit messy. Or is there a proposal how to deal with qt6 without introducing too much issues for conda-forge?

**Comment by hmaarrfk** [comment]
you can skip it if you want. but ffmpeg7 is quite new, and many projects don't always support it right off the bat.

you can add a `skip: true  # [ffmpeg == '6']` or something like that and rerender.

or just build the two and forget about it ;)

**Comment by looooo** [comment]
@hmaarrfk I just looked into setting up azure for my personal repo, but I am a bit lost (everything seems to have changed there). What do you think about setting up a qt6 branch for the vtk-feedstock and publish the builds to a qt6-label?

**Comment by hmaarrfk** [comment]
I would not setup a branch unless it has a different version number for VTK.

It increases the chance of a clash.

I've been using qt6 for a while now, i think it is OK.

i would keep it simple for yourself.

Azure kinda broke my personal repo workflow, need time to get it back up and running.

**Comment by looooo** [comment]
@hmaarrfk so there is nothing special needed for qt6 (for example renaming the builds as we did here: https://github.com/conda-forge/soqt-feedstock/blob/main/recipe/meta.yaml#L1L8)?

**Comment by hmaarrfk** [comment]
you added a constraint since the two packages clash:

https://github.com/conda-forge/soqt-feedstock/blob/main/recipe/meta.yaml#L40



OpenCV currently builds 3 variants:

* qt5
* qt6
* noqt

I'm trying to get rid of qt5 given the new version bump (easier to communicate) https://github.com/conda-forge/opencv-feedstock/pull/417

so given that you have a version bump here, might as well only build for qt6 and move on.

**Comment by looooo** [comment]
the osmesa build fails with:
```
_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include/GL/osmesa.h:323:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'OSMesaPostprocess'
```

**Comment by looooo** [comment]
@hmaarrfk is it ok to skip the osmesa builds for now?

**Comment by hmaarrfk** [comment]
That's between you and the other maintainers. 

Give them a week or two to respond or something?

**Comment by looooo** [comment]
continued in #328

**Pull request closed without merging**